### DISCUSSION TITLE
Universal Schema

### DISCUSSION THREAD

**Discussion by uma-uc** [vip-git/react-jsonschema-form-material-ui#417] [Q&A] [answered]
I am looking to write a universal schema to be used in svelte project.

To that end -  I wanted to better understand the dependency. 

- @react-jsonschema-form-interceptor
- @react-jsonschema-form-utils 

and if any porting to svelte will have to do. 

Where are these dependencies being hosted? I am not able to determine the source rep for the above.

Thanks in advance

**Comment by vip-git** [comment, marked as answer]
> I am looking to write a universal schema to be used in svelte project.
> 
> To that end - I wanted to better understand the dependency.
> 
> * @react-jsonschema-form-interceptor
> * @react-jsonschema-form-utils
> 
> and if any porting to svelte will have to do.
> 
> Where are these dependencies being hosted? I am not able to determine the source rep for the above.
> 
> Thanks in advance

Hi @uma-uc ,

Currently [universal schema](https://github.com/vip-git/react-jsonschema-form-material-ui/blob/main/universal-json-schema.md) is in its beta phase. 

I will shortly create a roadmap and timelines for each framework support to be included. I assume by end of this month you can expect a release for svelte and other major frameworks.

If you are interested in contributing to this project to add support for svelte then that would be amazing incase not then i reckon to wait for end of this month for more updates.

Thanks for your question :)

**Reply by uma-uc** [reply]
I will be happy to do my bit to contribute to svelte. Let me know

**Reply by vip-git** [reply]
Hi @uma-uc Thats great - Give me couple of days to merge the pending Universal Schema PR which outlines all the information needed to setup support for cross framework compilation. 

I will tag you in this PR so you have detailed knowledge on the same. 

Thanks for the patience on this - i will keep you posted.

**Reply by uma-uc** [reply]
perfect. Thanks - I look forward to it.. and thanks for the awesome package

**Reply by vip-git** [reply]
Hi @uma-uc

There is a barebone version merged which allows now cross framework compilation . I reckon to have a svelte starting 0.1 version with basic components you would need to wait for some time.

You can always give a try and see what you can make of it with the barebone version.

I will tag you in upcoming releases for svelte so you can give that a try as well.

Thanks - V

**Reply by uma-uc** [reply]
Thanks - I will - btw - I was thinking of using svelte-atoms - and extend the components for any missing ones. let me know your preferred library

**Reply by vip-git** [reply]
Yes that can go in as a UI Framework support in `svelte` my plan is to support for all `frameworks` a standard `Material UI` implementation.

In svelte case this will default to `sveltematerialui` but there is no stopping at supporting other UI Frameworks as the current setup allows you to scale to any Framework / Platform with integration with any UI Framework possible.

**Reply by uma-uc** [reply]
Hi @vip-git  - any updates? Thanks a lot

**Reply by vip-git** [reply]
Hi @uma-uc yes i tagged you in the recent release did you get a chance to try it ?
